app/models/user_option.rb

```diff
@@ -2,13 +2,15 @@
 
 class UserOption < ActiveRecord::Base
   HOMEPAGES = {
+    # -1 => reserved for "custom homepage"
     1 => "latest",
     2 => "categories",
     3 => "unread",
     4 => "new",
     5 => "top",
     6 => "bookmarks",
     7 => "unseen",
+    # 8 => reserved for "hot"
```

Comment:
Why are -1 and 8 commented out? 🤔

Reply:
Because they are kinda special. It's not ideal, but the `homepage` method returns the stored value, and both `-1` and `8` don't always work. 

https://github.com/discourse/discourse/blob/main/app/models/user_option.rb#L184-L188

`-1` we never store it, we only use it when a site has a theme with a custom homepage, and when the UserUpdater sees that value, it resets. (The use case here is user picking a homepage other than custom and then changing their mind, in that scenario we want to just reset their choice.) 

`8` is reserved for hot, but only used when the top_menu setting has "hot". (Arguably, that's not the best setup, but I didn't want to make any unrelated changes, so I just added the comment for clarity.)